src/io/export_3mf_v2.cc

```diff
@@ -121,6 +121,26 @@ static bool append_polyset(std::shared_ptr<const PolySet> ps, Lib3MF::PWrapper&
       }
     }
 
+    Lib3MF::PBaseMaterialGroup pBaseMaterial = model->AddBaseMaterialGroup();
+
+    std::vector<Lib3MF_uint32> ids;
+    for(int i=0;i<out_ps->colors.size();i++) {
+      sLib3MFColor col;
+      std::string colname="Color" + i;
+      Lib3MF_uint32 nBaseMaterialID = pBaseMaterial->AddMaterial(colname.c_str(), wrapper->FloatRGBAToColor(
+		out_ps->colors[i][0], out_ps->colors[i][1], out_ps->colors[i][2], out_ps->colors[i][3]));
+      ids.push_back(nBaseMaterialID);
+    }
+    if(ids.size() > 0) mesh->SetObjectLevelProperty(pBaseMaterial->GetResourceID(), ids[0]);
+    sLib3MFTriangleProperties tri_prop;
+    tri_prop.m_ResourceID=pBaseMaterial->GetResourceID();
+    for(int i=0;i<out_ps->color_indices.size();i++) {
+      tri_prop.m_PropertyIDs[0]=ids[out_ps->color_indices[i]];
+      tri_prop.m_PropertyIDs[1]=ids[out_ps->color_indices[i]];
+      tri_prop.m_PropertyIDs[2]=ids[out_ps->color_indices[i]];
```

Comment:
`PolySet.h`mentions that the color index can be -1, I think in that case we need to skip adding the property.

> https://github.com/openscad/openscad/blob/master/src/geometry/PolySet.h#L24

Reply:
Another reason to add a test :)